website/content/docs/install-boundary/terraform-patterns.mdx

```diff
@@ -0,0 +1,643 @@
+  scope_id    = boundary_scop.project.id 
+  plugin_name = "aws"
+
+  # This is the name of the AWS S3 bucket that will store the recordings
+  bucket_name = "mybucket1"
+
+  # the role_arn value should be the same arn used as the instance profile that is attached to 
+  # your Boundary Worker ec2 instance
+  # https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2_instance-profiles.html
+  attributes_json = jsonencode({
+    "region"                      = "us-east-1"
+    "role_arn"                    = "arn:aws:iam::123456789012:role/S3Access"
+    "disable_credential_rotation" = true
+  })
+}
+```
+At this point you have configured the pre-requisites for session recording. The last task is to
+actually enable recording on a target which is demonstrated in the Targets section below. 
+
+Please refer to the Terraform provider documentation for more information.
+[Storage Buckets](https://registry.terraform.io/providers/hashicorp/boundary/latest/docs/resources/storage_bucket)
+[Storage Policies](https://registry.terraform.io/providers/hashicorp/boundary/latest/docs/resources/policy_storage)
+## Targets
+Now that you have your hosts in a host catalog and have a credential store, you 
+can create [targets](/docs/concepts/domain-model/targets).
+
+This pattern is a simple ssh connection using an injected username and password.
+```hcl
+resource "boundary_target" "ssh_foo" {
+  name         = "ssh_foo"
+  description  = "Ssh target"
+  scope_id     = boundary_scope.project.id
+
+  # How the target connects
+  type         = "ssh"
+  default_port = "22"
+
+  # The hosts that this target can connect to
+  host_source_ids = [
+    boundary_host_set.foo.id
+  ]
+
+  # The credentials we will use to connect
+  injected_application_credential_source_ids = [
+    boundary_credential_library_vault.example.id
+  ]
+
+  # Add the next two lines to an existing target to enable session recording.
+  enable_session_recording = true
+  storage_bucket_id        = boundary_storage_bucket.aws_bucket.id
+}
+
+```
+
+Here is the same pattern but using brokered credentials.
+```hcl
+resource "boundary_target" "ssh_foo" {
+  name         = "ssh_foo"
+  description  = "Ssh target"
+  scope_id     = boundary_scope.project.id
+
+  # How the target connects
+  type         = "ssh"
+  default_port = "22"
+
+  # The hosts that this target can connect to
+  host_source_ids = [
+    boundary_host_set.foo.id
+  ]
+
+  # The credentials we will use to connect.
+  # Note that we use the static credential library created earlier
+  brokered_application_credential_source_ids = [
+    boundary_credential_library.example.id
+  ]
+
+   # Add the next two lines to an existing target to enable session recording.
+  enable_session_recording = true
+  storage_bucket_id        = boundary_storage_bucket.aws_bucket.id
+}
+```
+
+To create a target that connects to Windows servers using RDP, the pattern is very similar:
+```hcl
+resource "boundary_target" "ssh_foo" {
```

Comment:
Assuming this and the below name & description be changed to rdp_foo/rdp target?